doc/streaming.md

```diff
@@ -138,4 +138,39 @@ handle_x_log_data(StartLSN, EndLSN, Data, CbState) ->
     then you do not have any risk to lose data. 
     
     Otherwise (if you do not load all data from tables during erlang app startup) 
-    it is not recommended to set align_lsn to true. In this case to stop PG server stop epgsql replication first.
\ No newline at end of file
+    it is not recommended to set align_lsn to true. In this case to stop PG server stop epgsql replication first.
+    
+## Flow control
+
+It is possible to set `{active, N}` on a TCP or SSL (since OTP 21.3) socket. E.g. for SSL:
+```erlang
+Opts = #{ host => "localhost"
+        , username => "me"
+        , password => "pwd"
+        , database => "test"
+        , ssl => true
+        , ssl_opts => [{active, 10}]
+        },
+{ok, Conn} = epgsql:connect(Opts).
+```
+
+It is currently allowed only in the replication mode. Its main purpose is to control the flow of
+replication messages from Postgresql database. If a database is under a high load and a process, which
+handles the message stream, cannot keep up with it then setting this option gives the handling process
+ability to get messages on-demand.
+
+When the connection is in the asynchronous mode, a process which owns a connection will receive
+```erlang
+{epgsql, Connection, socket_passive}
+```
+The process decides when to activate connection's socket again. To do that it should call:
```

Comment:
```suggestion
as soon as underlying socket received N messages from network.

The client process decides when to activate connection's socket again. To do that it should call:
```

Reply:
I think that "messages" would be ambiguous here. It will receive `N` network frames of a size of current `MTU`...

Reply:
"N network frames of a size of current MTU" is not guaranteed either. By messages I mean rather Erlang messages, not network messages or Postgres protocol messages.

Maybe rephrase as "received N messges from `gen_tcp`/`ssl`, see https://www.erlang.org/doc/man/inet.html#setopts-2 `active` option description"?